README.md

```diff
@@ -1,28 +1,168 @@
+
+A very simple starter example, which just outputs a diagnosis:
+```py
+import ocptv.output as tv
+
+def main():
+    run = tv.TestRun(name="simple_diagnosis", version="1.0")
+    dut = tv.Dut(id="server0", name="server0.domain.net")
+    with run.scope(dut=dut):
+        step = run.add_step("check_fanspeed")
+        with step.scope():
+            if get_fan_speed() > 1600: # assuming external get_fan_speed() impl
+                step.add_diagnosis(DiagnosisType.PASS, verdict="fan_ok")
+            else:
+                step.add_diagnosis(DiagnosisType.FAIL, verdict="fan_low")
+```
+
+Expected output (slightly reformatted for readability):
+```json
+{"schemaVersion": {"major": 2, "minor": 0}, "sequenceNumber": 0, "timestamp": "2023-04-19T20:54:05.652514+01:00"}
+
+{"testRunArtifact": {
+    "testRunStart": {
+        "name": "simple_diagnosis",
+        "version": "1.0", "commandLine": "", "parameters": {},
+        "dutInfo": {
+            "dutInfoId": "server0", "name": "server0.domain.net",
+            "platformInfos": [], "softwareInfos": [], "hardwareInfos": []
+        }
+    }
+}, "sequenceNumber": 1, "timestamp": "2023-04-19T20:54:05.652646+01:00"}
+
+{"testStepArtifact": {
+    "testStepId": "0",
+    "testStepStart": { "name": "check_fanspeed" }
+}, "sequenceNumber": 2, "timestamp": "2023-04-19T20:54:05.652771+01:00"}
+
+{"testStepArtifact": {
+    "testStepId": "0",
+    "diagnosis": {
+        "verdict": "fan_ok", "type": "PASS",
+        "sourceLocation": {"file": "/home/user/ocp-diag-python/test.py", "line": 16}
+    }
+}, "sequenceNumber": 3, "timestamp": "2023-04-19T20:54:05.653037+01:00"}
+
+{"testStepArtifact": {
+    "testStepId": "0",
+    "testStepEnd": {"status": "COMPLETE"}
+}, "sequenceNumber": 4, "timestamp": "2023-04-19T20:54:05.653167+01:00"}
+
+{"testRunArtifact": {
+    "testRunEnd": {"status": "COMPLETE", "result": "PASS"}
+}, "sequenceNumber": 5, "timestamp": "2023-04-19T20:54:05.653219+01:00"}
+```
+
+For more examples of usage, there are a number available in the [examples folder](https://github.com/opencomputeproject/ocp-diag-python/tree/dev/examples), along with expected outputs.
+
+### Configuration
+
+There are a couple of knobs that can be used to configure the behavior of the `ocptv` library:
+- **enable_runtime_checks**: when this is true, the lib code will try to validate the actual types of the data being fed to it against what the specification requires. This may be disable if performance is critical or if it shows any false positives. Default is `True`.
+- **timezone**: a `datetime.tzinfo` implementation overriding the default local timezone. Default is `None`, which means local timezone.
+- **writer**: a `ocptv.output.Writer` implementation that is used for the lowlevel writing of serialized JSON strings. Default is `ocptv.output.StdoutWriter` which just writes everything to standard output.
+
+To setup any of these aspects:
+```py
+import ocptv.output as tv
+
```

Comment:
The import for FileWriter is missing

Reply:
ok, so it wasnt obvious that `FileWriter` is usercode. i'll replace it with a `StdoutWriter`. though i could also argue that these should come with the lib, incl rolling file; i'll leave this as a feature request